### PR TITLE
[P1-L01] Compute late penalty based on current time rather than endTime

### PR DESCRIPTION
**Pull request by mrice32**
This updates the store to charge the late penalty based on the `currentTime` rather than the `endTime` passed in.

As a part of this change, the weekly delay fee is now charged on a per-second basis since `startTime - endTime` no longer has time constraints. This means that weekly delay fees should be scaled down by `604800` (number of seconds in a week). If we didn't do this, a contract that splits a period into two calculate calls would pay twice the late penalty that a contract that pays it all in one call would pay.

Many tests had to be updated to accommodate these changes, since this means that `startTime` must always be <= `currentTime`.

### CODE REVIEW

**Comment by ptare** [comment]
IIUC we use the current time to determine the late fee and the passed in end time to calculate the regular fee. When would the passed in end time be different from the current time? Maybe in some weird situations post expiry?

**Comment by mrice32** [comment]
> IIUC we use the current time to determine the late fee and the passed in end time to calculate the regular fee. When would the passed in end time be different from the current time? Maybe in some weird situations post expiry?

@ptare yeah, post expiry is one example. I think that's probably the only one that exists in our current contracts.

File: core/test/financial-templates/Liquidatable.js [hunk 1]
```diff
@@ -1363,27 +1368,30 @@ contract("Liquidatable", function(accounts) {
       // Update the liquidatableParameters to use the new token as collateral and deploy a new Liquidatable contract
       let USDCLiquidatableParameters = liquidatableParameters;
       USDCLiquidatableParameters.collateralAddress = collateralToken.address;
-      USDCLiquidationContract = await Liquidatable.new(USDCLiquidatableParameters, {
```
Comment (chrismaree) [inline core/test/financial-templates/Liquidatable.js:1366]: +1, unrelated to PR but good cleanup.

Comment (mrice32) [reply core/test/financial-templates/Liquidatable.js:1366]: Yeah, it was way easier to build a function that updates the time in the store and the liqudiatable contract if the liquidatable contract was always called the same thing.

**Comment by mrice32** [comment]
Since #1236 unified all the timing logic, I was able to substantially cut the scope of the test changes required for this PR. PTAL.

File: core/migrations/8_deploy_store.js [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
-const { getKeysForNetwork, deploy } = require("../../common/MigrationUtils.js");
+const { getKeysForNetwork, deploy, enableControllableTiming } = require("../../common/MigrationUtils.js");
 const { interfaceName } = require("../utils/Constants.js");
 
 module.exports = async function(deployer, network, accounts) {
   const keys = getKeysForNetwork(network, accounts);
+  const controllableTiming = enableControllableTiming(network);
 
-  const { contract: store } = await deploy(deployer, network, Store, { from: keys.deployer });
+  const { contract: store } = await deploy(deployer, network, Store, Timer.address, { from: keys.deployer });
```
Comment (nicholaspai) [inline core/migrations/8_deploy_store.js:12]: I believe this should address(0) iff `controllableTIming == false`

File: core/test/financial-templates/Liquidatable.js [hunk 3]
```diff
@@ -91,6 +90,10 @@ contract("Liquidatable", function(accounts) {
   };
 
   beforeEach(async () => {
+    // Force each test to start with a simulated time that's synced to the startTimestamp.
+    const timer = await Timer.deployed();
+    await timer.setCurrentTime(startTime);
```
Comment (nicholaspai) [inline core/test/financial-templates/Liquidatable.js:95]: This is a better solution to creating a new Timer, to instead just set the time at the beginning of each test.

**Pull request merged**